### PR TITLE
docs: update OAuth2 getting started doc

### PR DESCRIPTION
**Pull request by tomekpapiernik**
This PR updates the OAuth2 federation guide:

- Authorization Code Grant now runs using Ory Elements - no need to run anything locally
- Old instructions repurposed to show how to tie custom UI into the flow
- Added instructions for Client Credentials Grant
- New SEO `title` and doc title (`h1`)
- Other tweaks

## Related Issue or Design Document

https://github.com/ory-corp/cloud/issues/3325

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by tomekpapiernik** [comment]
`format (pull_request)` fail unrelated to these changes

**Comment by aeneasr** [comment]
Can we move the custom ui to a separate document? I think we should keep the intro as easy as pie :)

**Comment by tomekpapiernik** [comment]
> Can we move the custom ui to a separate document? I think we should keep the intro as easy as pie :)

There's very little information about the custom UI other than the configuration instructions so IMO it's not standalone doc material. I could try to make it even shorter if you think this complicates the message too much? 

Please remember about the right-hand side navigation which shows all of the doc content from the get-go, so the users can read the overview and jump straight to the grants that they're interested in. This very section is easy to skip altogether if it's not relevant for the reader. ;)

**Comment by kmherrmann** [comment]
But now we have details on how to configure a custom UI but removed the part about actually running the consent node, which i believe is confusing and needs clearing up. If we keep it, we need to clarify this ("in case you are writing your own user interface... this is how you configure it) and maybe put it into a "note" box.

Also - can we have screenshots of the end user flow? :)

**Comment by aeneasr** [comment]
> Also - can we have screenshots of the end user flow? :)

You can re-use the video from here: https://www.ory.sh/run-oauth2-server-open-source-api-security/#validate-the-oauth2-access-token

> There's very little information about the custom UI other than the configuration instructions so IMO it's not standalone doc material. I could try to make it even shorter if you think this complicates the message too much?

Re-add that documentation on running the node app etc, and make it into a separate guide. It does not belong in the "get this stuff running in 2 minutes" guide - this should focus on: copy these 3 commands and it works.

**Comment by tomekpapiernik** [comment]
@aeneasr @kmherrmann  Updated the PR. The custom UI is now relegated to a separate document in hydra/guides, added the auth code grant video to the doc in getting started.

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 1]
```diff
@@ -1,248 +1,187 @@
...
-
-By following this guide you will:
-
-- Create an OAuth2 client in The Ory Network.
-- Run a simple login and consent Node.js application that allows the user to grant or reject access to their data on your machine.
-- Configure The Ory Network to work with the sample application that runs on your machine.
-- Perform Authorization Code Grant in The Ory Network using the login and consent app and a sample web server acting as an OAuth2
-  Client.
+- Creating OAuth2 clients in The Ory Network with the Ory CLI
+- Using the Authorization Code Grant (with OpenID) flow with user accounts created in Ory Identities and login and consent screens
```
Comment (piotrmsc) [inline docs/getting-started/ory-cloud-oauth2.mdx:17]: I would phrase it more in a way : federation to Ory Identity Server with consent screen supplied by Ory Account Experience

**Review by aeneasr — DISMISSED** [review]
Nice, this looks good to me now :)

File: docs/getting-started/ory-cloud-oauth2.mdx [hunk 2]
```diff
@@ -1,248 +1,187 @@
...
-
-Provide this data in the cURL request:
+:::
 
-- `SDK_CONFIGURATION_URL` - found in the 'Connect' section of the Ory Console. Must end with '/admin/clients'.
-- `ORY_PAT`: Ory Personal Access Token copied from the Ory Console, 'ory_pat' prefix included.
+## Authorization Code Grant
 
-:::
+The [Authorization Code Grant](https://www.rfc-editor.org/rfc/rfc6749#section-1.3.1) is most commonly used in scenarios where
```
[the remark this answers is not in the document]
Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx]: done

File: docs/getting-started/ory-cloud-oauth2.mdx (hunk 1 above)
Comment (tomekpapiernik) [reply docs/getting-started/ory-cloud-oauth2.mdx]: done

**Review by aeneasr — APPROVED** [review]

**Pull request merged**